.github/workflows/build.yml

```diff
@@ -29,7 +29,15 @@ concurrency:
   group: ${{github.ref}}
   cancel-in-progress: true
 jobs:
+  start_linux_ec2_runner:
+    uses: ./.github/workflows/ec2_runner_jobs.yml
+    secrets: inherit
+    permissions: write-all
+    with:
+      job_type: start
+
   common_config:
+    needs: [start_linux_ec2_runner]
```

Comment:
I presume the EC2 instance will take a while to start. Is it really necessary to block the entire build?

It's better to pass the `distro` as a separate `workflow_call.input` to `build_steps`. It will also avoid having to duplicate all the matrix settings below, which might cause confusion.